### PR TITLE
bem-xjst: support previous source maps

### PR DESCRIPTION
**Pull request by miripiruni**
fix for #283

cc @sbmaxx 

Related: 
 * https://github.com/enb/enb-source-map/pull/19
 * https://github.com/bem/bem-xjst/pull/491

### CODE REVIEW

File: techs/bem-xjst.js [hunk 1]
```diff
@@ -119,18 +107,29 @@ module.exports = buildFlow.create()
...
-                // join source code
-                codeToCompile = sources.map(function (source) {
-                    return source.contents;
-                }).join(EOL);
+                file = new File(this.node.resolvePath(this._target), { sourceMap: this._sourcemap }),
+                needWrapIIFE = this._iife;
+
+            sources.forEach(function (source) {
+                file.writeLine('/* begin: ' + source.path + ' */');
+                needWrapIIFE && file.writeLine('(function(){');
```
Comment (sbmaxx) [inline techs/bem-xjst.js]: yeaah! extra like for iife support :)

**Comment by miripiruni** [comment]
@tadatuta @zxqfox review needed

File: techs/bem-xjst.js [hunk 2]
```diff
@@ -1,8 +1,8 @@
-var EOL = require('os').EOL,
-    vow = require('vow'),
+var vow = require('vow'),
     enb = require('enb'),
     vfs = enb.asyncFs || require('enb/lib/fs/async-fs'),
     buildFlow = enb.buildFlow || require('enb/lib/build-flow'),
+    File = require('enb-source-map/lib/file'),
```
[the remark this answers is not in the document]
Comment (miripiruni) [reply techs/bem-xjst.js]: :—(

**Comment by tadatuta** [comment]
Please update README according to changes

File: techs/bemhtml.js [hunk 3]
```diff
@@ -53,6 +53,8 @@ module.exports = require('./bem-xjst').buildFlow()
     .defineOption('naming')
     .defineOption('engineOptions')
     .defineOption('requires', {})
+    .defineOption('iife', false)
```
Comment (tadatuta) [inline techs/bemhtml.js]: I've got two questions about this option:
1. Maybe we can provide more generic solution like `wrap: { before: 'str', after: 'str' }` and even take a custom function to wrap initial content (see https://github.com/tadatuta/enb-wrap for reference)?
2. shouldn't it be introduced with separate PR or at least separate commit?

Comment (miripiruni) [reply techs/bemhtml.js]: @tadatuta 

1. why we might need more generic solution here?
2. OK, let’s split into two commits.

Comment (sbmaxx) [reply techs/bemhtml.js]: we really need iife & source maps fast :) can't live without them

**Comment by zxqfox** [comment]
Great!

**Comment by miripiruni** [comment]
+ enb-bemxjst@8.9.0
